changelog/9131.misc.md

```diff
@@ -0,0 +1 @@
+Change `TrainingDataImporter` to be sync.
```

Comment:
Can we please also update the docs [here](https://rasa.com/docs/rasa/training-data-importers) and add a migration guide entry? 🙌🏻